crates/api/src/module.rs

```diff
@@ -131,17 +126,19 @@ impl Module {
         // Note that the call to `unsafe` here should be ok because we
         // previously validated the binary, meaning we're guaranteed to pass a
         // valid binary for `store`.
-        unsafe { Module::new_unchecked(store, binary) }
+        unsafe { Module::new_internal(store, binary, None) }
     }
 
     /// Creates a new WebAssembly `Module` from the given in-memory `binary`
     /// data. The provided `name` will be used in traps/backtrace details.
     ///
     /// See [`Module::new`] for other details.
     pub fn new_with_name(store: &Store, binary: &[u8], name: &str) -> Result<Module> {
-        let mut ret = Module::new(store, binary)?;
-        Rc::get_mut(&mut ret.inner).unwrap().name = Some(name.to_string());
```

Comment:
I may be missing something, but how come this refactor was necessary?

Reply:
Ah, this is because:

1. Former version had a cascade like initialization: `new_with_name` called `new` which called `new_unchecked`. 
2. They are all public therefore each of them can be called independently.
3. Compilation requires supplying the name of the module.
4. And as discussed before, a constructor must return a `Module` with already compiled module.

Reply:
Oh right sorry, I forgot that compilation required the resolved name of the module. Ok, we can always try to clean this up later if necessary!